app/javascript/components/test-utils.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+
+
+// a helper method for invoking a class method (for unit tests)
+// obj = a class
+// func = a tested function
+// objThis = an object's this
+// arg = function args
+export const classFunctionUnitTest = (obj, func, objThis, args) =>
+  obj.prototype[func].apply(objThis, args);
+
+/**
+ * Shallow render a component multipile times with fixtures
+ * @param  {ReactComponent} Component Component to shallow-render
+ * @param  {Object}         fixtures  key=fixture description, value=props to apply
+ * @return {Object}                   key=fixture description, value=shallow-rendered component
+ */
+export const shallowRenderComponentWithFixtures = (Component, fixtures) =>
+  Object.entries(fixtures).map(([description, props]) => ({
+    description,
+    component: shallow(<Component {...props} />),
+  }));
+
+/**
+ * Test a component with fixtures and snapshots
+ * @param  {ReactComponent} Component Component to test
+ * @param  {Object}         fixtures  key=fixture description, value=props to apply
+ */
+export const testComponentSnapshotsWithFixtures = (Component, fixtures) =>
+  shallowRenderComponentWithFixtures(Component, fixtures).forEach(({ description, component }) =>
+    it(description, () => expect(toJson(component)).toMatchSnapshot()));
+
+/**
+ * run an action (sync or async) and except the results to much snapshot
+ * @param  {Function}  runAction  Action runner function
+ * @return {Promise}
+ */
+export const testActionSnapshot = async(runAction) => {
```

Comment:
(this one **is** used, in `app/javascript/spec/ansibleCatalog/extraVarRow.test.jsx`)